### PR TITLE
RESTWS-716 : Order Resource should Return Exceptioninstead of Empty R…

### PR DESCRIPTION
**Pull request by tendomart**
https://issues.openmrs.org/browse/RESTWS-716
@dkayiwa  @mogoodrich  

can you please look into this ?

### CODE REVIEW

**Comment by tendomart** [comment]
I looked into the Travis logs ,am just wondering is their a way to track all the classes methods utilizing "EmptyResult.toSimpleObject() " ? The one i altered (OrderResource) passes the test ,but am not yet sure where else EmptyResult.toSimpleObject() is utilised

File: omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java [hunk 1]
```diff
@@ -485,7 +486,12 @@ public void shouldGetAllInActiveOrdersForAPatient() throws Exception {
 		        PropertyUtils.getProperty(resultList.get(1), "uuid").toString() });
 		assertThat(uuids, hasItems(expectedOrderUuids));
 	}
-	
+	@Test
+	public void shouldReturnExceptionIfNoUuidIsSpecified() throws Exception {
```
Comment (samuelmale) [inline omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java]: Can you improve the name of the test? eg. `shouldThrowExceptionIfPatientQueryParameterNotSpecified`

Comment (tendomart) [reply omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java]: Ok thanks alot for that revision.

Comment (tendomart) [reply omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java]: Ok thanks alot for that revision.

**Comment by tendomart** [comment]
> > ,but am not yet sure where else EmptySearchResult.toSimpleObject() is utilised
> 
> Is this search query of help https://github.com/openmrs/openmrs-module-webservices.rest/search?q=toSimpleObject&unscoped_q=toSimpleObject?

Yes it has been an awesome insight  , Thanks Samuel,though i still need toSimpleObject(..) from 
  EmptySearchResult ,however it looks like **EmptySearchResult** is bot used in many places...I was wondering which other instances were affected by my change.

**Comment by dkayiwa** [comment]
@tendomart did you figure out the cause of travis failure?

**Comment by tendomart** [comment]
@dkayiwa  Nop. I only saw failing tests other classes but no root cause .

**Comment by dkayiwa** [comment]
Are you still resolving issues?

**Comment by tendomart** [comment]
@dkayiwa  yes i re-run all tests locally and saw errors because of the parameter i introduced.Sorry for taking longer than expected , am multi - tasking.Still trying out other OpenMRS native Exceptions and hot to throw them without affecting  methods with PageableResult return types. There are still Type Casting confilcts against any **Collections** sub-class **instances**.

**Comment by dkayiwa** [comment]
@tendomart are you still working on this?

**Comment by tendomart** [comment]
Yes Sir still working.

**Comment by mozzy11** [comment]
@tendomart , do the tests run successful on your local machine before you push?? or else  you would need to ensure every thing runs fine @Travis falure

**Comment by tendomart** [comment]
Yes Sir still working.

**Comment by tendomart** [comment]
@mozzy11  they did for some Instances ,but not golbally

**Comment by mozzy11** [comment]
alryt. did you ensure they run globally before you made any local change?

**Comment by dkayiwa** [comment]
@tendomart did you abandon this?

**Comment by dkayiwa** [comment]
Closing because the author is not working on it any more.

**Pull request closed without merging**